File: web_app/backend/pillmate/src/main/java/com/example/pillmate/model/Pharmacy.java
package com.example.pillmate.model;


import java.util.Date;


public class Pharmacy {
    private String pharID;
    private String storeName;
//    private Date date;
    private String address;
    private String province;
    private String city;
    private String latitude;
    private String longitude;
    private String phoneNumber;
    private String serviceTime;
    private String serviceDate;

    private String email;

    public Pharmacy(){

    }

    public Pharmacy(String pharID, String storeName, String address, String latitude, String longitude, String phoneNumber, String serviceTime, String serviceDate, String province, String city, String email){
        this.pharID = pharID;
        this.storeName = storeName;
        this.address = address;
        this.latitude = latitude;
        this.longitude = longitude;
        this.phoneNumber = phoneNumber;
        this.serviceTime = serviceTime;
        this.serviceDate = serviceDate;
        this.province = province;
        this.city = city;
        this.email = email;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getCity() {
        return city;
    }

    public String getProvince() {
        return province;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public void setProvince(String province) {
        this.province = province;
    }

    public String getPharID() {
        return pharID;
    }

    public String getAddress() {
        return address;
    }

    public String getLatitude() {
        return latitude;
    }

    public String getLongitude() {
        return longitude;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public String getServiceDate() {
        return serviceDate;
    }

    public String getServiceTime() {
        return serviceTime;
    }

    public String getStoreName() {
        return storeName;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public void setLatitude(String latitude) {
        this.latitude = latitude;
    }

    public void setLongitude(String longitude) {
        this.longitude = longitude;
    }

    public void setPharID(String pharID) {
        this.pharID = pharID;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public void setServiceDate(String serviceDate) {
        this.serviceDate = serviceDate;
    }

    public void setServiceTime(String serviceTime) {
        this.serviceTime = serviceTime;
    }

    public void setStoreName(String storeName) {
        this.storeName = storeName;
    }
}

File: web_app/frontend/pillmate-app/src/models/Pharmacy.ts
class PharmacyModel {
    pharID: string;
    storeName: string;
    address: string;
    province: string;
    city: string;
    latitude: string;
    longitude: string;
    phoneNumber: string;
    serviceTime: string;
    serviceDate: string;
    email: string;

    constructor() {
        this.pharID = '';
        this.storeName = '';
        this.address = '';
        this.province = '';
        this.city = '';
        this.latitude = '';
        this.longitude = '';
        this.phoneNumber = '';
        this.serviceTime = '';
        this.serviceDate = '';
        this.email ='';
    }

    setPharID(value: string){
        this.pharID = value;
    }

    setStoreName(value: string) {
        this.storeName = value;
    }

    setAddress(value: string) {
        this.address = value;
    }

    setProvince(value: string) {
        this.province = value;
    }

    setCity(value: string) {
        this.city = value;
    }

    setLatitude(value: string) {
        this.latitude = value;
    }

    setLongitude(value: string) {
        this.longitude = value;
    }

    setPhoneNumber(value: string) {
        this.phoneNumber = value;
    }

    setServiceTime(value: string) {
        this.serviceTime = value;
    }

    setServiceDate(value: string) {
        this.serviceDate = value;
    }

    setEmail(value:string){
        this.email = value;
    }
}

export default PharmacyModel;
